src/index.ts

```diff
@@ -302,61 +313,18 @@ export default class EditorjsList {
       },
     ];
 
-    if (this.listStyle === 'ordered') {
-      const startWithElement = renderToolboxInput(
-        (index: string) => this.changeStartWith(Number(index)),
-        {
-          value: String((this.data.meta as OrderedListItemMeta).start ?? 1),
-          placeholder: '',
-          attributes: {
-            required: 'true',
-          },
-          sanitize: input => stripNumbers(input),
-        });
-
-      const orderedListTunes: MenuConfigItem[] = [
-        {
-          label: this.api.i18n.t('Start with'),
-          icon: IconStartWith,
-          children: {
-            items: [
-              {
-                element: startWithElement,
-                // @ts-expect-error ts(2820) can not use PopoverItem enum from editor.js types
-                type: 'html',
-              },
-            ],
-          },
-        },
-      ];
-
-      const orderedListCountersTunes: MenuConfigItem = {
-        label: this.api.i18n.t('Counter type'),
-        icon: OlCounterIconsMap.get((this.data.meta as OrderedListItemMeta).counterType!),
-        children: {
-          items: [],
-        },
-      };
-
-      /**
-       * For each counter type in OlCounterType create toolbox item
-       */
-      OlCounterTypesMap.forEach((_, counterType: string) => {
-        orderedListCountersTunes.children.items!.push({
-          title: this.api.i18n.t(counterType),
-          icon: OlCounterIconsMap.get(OlCounterTypesMap.get(counterType)!),
-          isActive: (this.data.meta as OrderedListItemMeta).counterType === OlCounterTypesMap.get(counterType),
-          closeOnActivate: true,
-          onActivate: () => {
-            this.changeCounters(OlCounterTypesMap.get(counterType) as OlCounterType);
-          },
-        });
-      });
-      // @ts-expect-error ts(2820) can not use PopoverItem enum from editor.js types
-      defaultTunes.push({ type: 'separator' }, ...orderedListTunes, orderedListCountersTunes);
-    }
-
-    return defaultTunes;
+    if (EditorjsList.displayedEntries){
+      if (this.listStyle === 'ordered') this.addAdditionalTunes(defaultTunes);
+    } else if (this.listStyle === 'ordered') this.addAdditionalTunes(defaultTunes);
+
+    return EditorjsList.displayedEntries ? defaultTunes.filter(
+      tune => {
+          if ('title' in tune) {
+            if(tune.title === 'Start with' || tune.title === 'Counter type') return true;
+            else return EditorjsList.displayedEntries?.includes(tune.title!.toLowerCase() as ListDataStyle);
+          } else if (('type' in tune) && tune.type === 'separator') return true;
+        }
+    ) : defaultTunes;
```

Comment:
you can firstly filter default tunes, and then add additional ones, then you won't need to check for `Start with` and `Counter type` in tunes